src/internal/pjsdb/job_cache.go

```diff
@@ -87,4 +84,37 @@ func writeToJobCache(ctx context.Context, extCtx sqlx.ExtContext, job Job) error
 		return errors.New("no rows affected")
 	}
 	return nil
+
+}
+
+// the 'stub' is created on a cache miss or when the cache read is disabled for a specific job.
+func writeStubToJobCache(ctx context.Context, extCtx sqlx.ExtContext, job Job) error {
+	ctx = pctx.Child(ctx, "writeStubToJobCache")
+	return jobCacheWriteWrapper(ctx, extCtx, `
+		INSERT INTO pjs.job_cache (job_id, cache_read, cache_write) 
+		VALUES ($1, $2, $3)
+	`, job.ID, job.ReadEnabled, job.WriteEnabled)
+}
+
+// a full write is only done on a cache hit.
```

Comment:
I just reviewed a PR of Bob's static check of comments. The comments would be better starts with "writeToJobCache" does something (to pass the checks) 😀 . Same applies to other functions.